### PR TITLE
Archim2 USB Hang Fix

### PR DESCRIPTION
**Pull request by InsanityAutomation**
Archim2 Board fixes : S
Set pins needed for USB host based on the USB feature, not just the display.
Fix output error in USB host debug
Expose option to disable USB drive exposure to PC host, as shared resource hung the main loop.

### CODE REVIEW

File: Marlin/src/sd/usb_flashdrive/Sd2Card_FlashDrive.cpp [hunk 1]
```diff
@@ -161,16 +161,16 @@ void DiskIODriver_USBFlash::idle() {
...
+          SERIAL_ECHOLNPGM("IDLE");
+          SERIAL_ECHOLNPGM("RESET_DEVICE");
+          SERIAL_ECHOLNPGM("RESET_NOT_COMPLETE");
+          SERIAL_ECHOLNPGM("DEBOUNCE");
+          SERIAL_ECHOLNPGM("DEBOUNCE_NOT_COMPLETE");
+          SERIAL_ECHOLNPGM("WAIT_SOF");
+          SERIAL_ECHOLNPGM("ERROR");
+          SERIAL_ECHOLNPGM("CONFIGURING");
+          SERIAL_ECHOLNPGM("CONFIGURING_DONE");
+          SERIAL_ECHOLNPGM("RUNNING");
```
Comment (thinkyhead) [inline Marlin/src/sd/usb_flashdrive/Sd2Card_FlashDrive.cpp]: This seems to remove the logic for which string to print.

File: Marlin/src/pins/sam/pins_ARCHIM2.h [hunk 2]
```diff
@@ -237,7 +237,7 @@
 //
 // LCD / Controller
 //
-#if ANY(HAS_WIRED_LCD, TOUCH_UI_ULTIPANEL, TOUCH_UI_FTDI_EVE)
+#if ANY(HAS_WIRED_LCD, TOUCH_UI_ULTIPANEL, TOUCH_UI_FTDI_EVE, USB_FLASH_DRIVE_SUPPORT)
```
Comment (thinkyhead) [inline Marlin/src/pins/sam/pins_ARCHIM2.h]: Which of the LCD and encoder pins is required with `USB_FLASH_DRIVE_SUPPORT`? Is it just `SD_DETECT_PIN` that is wanted here?

File: Marlin/src/sd/usb_flashdrive/Sd2Card_FlashDrive.cpp (hunk 1 above)
Comment (InsanityAutomation) [reply Marlin/src/sd/usb_flashdrive/Sd2Card_FlashDrive.cpp]: Most of the items there were undefined, as it was not encapsulated strings, therefore enabling USB debug resulted in compile errors.

File: Marlin/src/pins/sam/pins_ARCHIM2.h (hunk 2 above)
Comment (InsanityAutomation) [reply Marlin/src/pins/sam/pins_ARCHIM2.h]: I believe 1 other pin was needed, but ill need to go back and check after the day job. SD Detect was def required though.

**Pull request merged**